controllers/nutanixmachine_controller.go

```diff
@@ -301,7 +317,7 @@ func (r *NutanixMachineReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 func (r *NutanixMachineReconciler) reconcileDelete(rctx *nctx.MachineContext) (reconcile.Result, error) {
 	ctx := rctx.Context
 	log := ctrl.LoggerFrom(ctx)
-	nc := rctx.NutanixClient
+	v3Client := rctx.NutanixClientV3
```

Comment:
nit: the PR may be a little easier to review if we kept `rctx.NutanixClient` and `nc` and just added the V4 suffix as needed.
As long as these are all the places where it needed to be changed, probably not worth changing back now though.

Reply:
i agree. keeping the changes low is required at such late stage.